src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
-
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            int buy = 0;
+            int supply = 0;
+            String line;
+            String[] data;
+            while ((line = reader.readLine()) != null) {
+                data = line.split(",");
+                switch (data[0]) {
+                    case "buy":
+                        buy += Integer.parseInt(data[1]);
+                        break;
+                    case "supply":
+                        supply += Integer.parseInt(data[1]);
+                        break;
+                    default:
+                }
+            }
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
```

Comment:
The BufferedWriter is still inside the try-with-resources block, which is executed after the loop. This means it will not overwrite the file multiple times, but it should still be outside the loop to ensure that the file is opened and written to only once after all data is processed. Consider moving the BufferedWriter outside the loop to improve efficiency and ensure all data is written correctly.